### PR TITLE
[BugFix] Fix the wrong result while evaluating conjuncts by slot in JNI connector

### PR DESCRIPTION
**Pull request by miomiocat**
Fixes #issue

## What type of PR is this:
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [x] 3.0
  - [x] 2.5
  - [ ] 2.4

### CODE REVIEW

File: be/src/exec/jni_scanner.h [hunk 1]
```diff
@@ -88,7 +88,6 @@ class JniScanner : public HdfsScanner {
 
     std::map<std::string, std::string> _jni_scanner_params;
     std::string _jni_scanner_factory_class;
-    Filter _chunk_filter;
```
Comment (dirtysalt) [inline be/src/exec/jni_scanner.h:91]: _chunk_filter could avoid allocation everytime you call it.

File: be/src/exec/hdfs_scanner.cpp [hunk 2]
```diff
@@ -360,18 +360,19 @@ void HdfsScannerContext::append_not_existed_columns_to_chunk(ChunkPtr* chunk, si
...
+            ASSIGN_OR_RETURN(chunk_size, ExecNode::eval_conjuncts_into_filter(it.second, chunk->get(), &filter))
             if (chunk_size == 0) {
-                break;
+                (*chunk)->set_num_rows(0);
+                return Status::OK();
             }
         }
-        if (chunk_size != 0 && chunk_size != (*chunk)->num_rows()) {
-            (*chunk)->filter(*filter);
+        if (chunk_size != 0 && chunk_size < (*chunk)->num_rows()) {
```
Comment (dirtysalt) [inline be/src/exec/hdfs_scanner.cpp:374]: is there any subtle difference between != and < ?

**Review by dirtysalt — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Pull request merged**